.github/workflows/_container.yml

```diff
@@ -11,7 +18,16 @@ jobs:
         with:
           # Need this to get version number from last tag
           fetch-depth: 0
-
+      - name: Validate PEP 440 version compliance
+        run: |
+          ref="${{ github.ref_name }}"
+          my_regex=${{env.PEP_440_REGEX}}
+          if [[ "$ref" =~ $my_regex ]]; then
+            echo "PEP 440 compliant version: $ref"
+          else
+            echo "Invalid PEP 440 version: $ref"
+            exit 1
+          fi
```

Comment:
These changes seem unrelated, why are they required?

Reply:
otherwise it rans into #411 , so this PR fixes both,

Reply:
This doesn't fix #411 because if the user makes a release with a wrongly-formatted tag, the release will still appear in [Github](https://github.com/DiamondLightSource/blueapi/releases) and PyPi(https://pypi.org/project/blueapi/#history). I don't think there's any harm in checking here, although it would be redundant if we could do something like [this](https://github.com/DiamondLightSource/blueapi/issues/411#issuecomment-2429132946).

However, PEP440 actually allows for a wider range of error messages than helm (for instance, it allows `1.1a1` where helm would require `1.1-a1`) so this check and error message aren't quite right. See https://helm.sh/docs/chart_best_practices/conventions/

Reply:
well we convert to semver later (      - name: Convert PEP 440 version to SemVer) , is that ok?

Reply:
I would be okay with remove the PEP440 check entirely and making a separate issue for adding SemVer check. I would not be okay with leaving the PEP440 check in since it won't catch any useful errors reported in #411.

Reply:
why not semver check in this PR?

Reply:
That would also be acceptable. It's only the PEP 440 check that I won't approve.